R/oa_fetch.R

```diff
@@ -586,7 +588,13 @@ oa_query <- function(filter = NULL,
     query = query
   )
 
-  if (verbose) message("Requesting url: ", query_url)
+  if (is.null(oa_print())){
+    url_display <- query_url
+  } else {
+    url_display <- paste0(substr(query_url, 1, oa_print()), "...")
```

Comment:
Can I make two suggestions for this?

1) Run the query through `URLdecode()` first for readability. If users want the debug the exact query string with special characters escaped, they can go back into "print everything" mode.
2) Only add ellipsis when there's more to the URL that's been cut out

```suggestion
  } else {
    query_url <- URLdecode(query_url)
    query_url_more <- if (oa_print() < nchar(query_url)) "..."
    url_display <- paste0(substr(query_url, 1, oa_print()), query_url_more)
```

Reply:
Great suggestions, June! Thank you!!! 💯